modules/tests/test/tekton/pipelinerun.go

```diff
@@ -0,0 +1,108 @@
+package tekton
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"sync"
+	"time"
+
+	"github.com/kubevirt/kubevirt-tekton-tasks/modules/tests/test/constants"
+	"github.com/kubevirt/kubevirt-tekton-tasks/modules/tests/test/framework/clients"
+
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	tkntest "github.com/tektoncd/pipeline/test"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/yaml"
+)
+
+type taskRunsLogs struct {
+	mu   sync.Mutex
+	logs map[string]string
+}
+
+func (l *taskRunsLogs) getLog(podName string) string {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	return l.logs[podName]
+}
+
+func (l *taskRunsLogs) setLog(podName, log string) {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	l.logs[podName] = log
+}
+
+func (l *taskRunsLogs) getAllLogs() string {
+	logs := ""
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	for podName, podLog := range l.logs {
+		logs += fmt.Sprintf("\n %s: %s", podName, podLog)
+	}
+	return logs
+}
+
+func WaitForPipelineRunState(clients *clients.Clients, namespace, name string, timeout time.Duration, inState tkntest.ConditionAccessorFn) (*v1beta1.PipelineRun, string) {
+	pipelinePodsLogs := taskRunsLogs{}
+	pipelinePodsLogs.logs = make(map[string]string)
+	var wg sync.WaitGroup
+	err := wait.PollImmediate(constants.PollInterval, timeout, func() (bool, error) {
+		pipelineRun, err := clients.TknClient.PipelineRuns(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+		if err != nil {
+			return true, err
+		}
+
+		for _, taskRun := range pipelineRun.Status.TaskRuns {
+			podName := taskRun.Status.PodName
+			if pipelinePodsLogs.getLog(podName) == "" && podName != "" {
+				wg.Add(1)
+				go func(pipelineNamespace, name string) {
+					defer wg.Done()
+					req := clients.CoreV1Client.Pods(pipelineNamespace).GetLogs(name, &v1.PodLogOptions{
+						Follow: true,
+					})
+					podLogs, err := req.Stream(context.TODO())
+					if err == nil {
+						defer podLogs.Close()
+						defer GinkgoRecover()
+
+						result, err := ioutil.ReadAll(podLogs)
+						pipelinePodsLogs.setLog(name, string(result))
+
+						gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
```

Comment:
What if `req.Stream` returns an error? Shouldn't that be checked too?

Reply:
In case of error, the function will be triggered again. In this case I don't thing it has some value to log error.

Reply:
I'm not sure about silently failing. It also ends the polling when getting the `PipelineRuns` fails. Also do we need to call `podLogs.Close` even if an error occured?

Reply:
Without pipelineRun the script can't do anything, that is why its failing. 

I don't think, that we need to fail. The pod can be e.g. in starting phase (and thus logs unavailable). But in next iteration the connection can be successfull. I can add some loging, but failing the whole script is not neccessary.

We don't have to call `podLogs.Close` when error occured, because the `req.Stream` fn returns always `nil, err`, so podLogs will always be nil, when `err != nil`

Reply:
Okay, then we can keep it like this. Maybe just add a comment so we know that it's okay to fail, like you wrote?

Reply:
I added the comment